### ISSUE TITLE
Breaking resin walls leaves behind darkened turfs

### ISSUE THREAD

**Issue by XElectricX** [tgstation/TerraGov-Marine-Corps#4021]
## Testmerges:

Master branch.

## Reproduction:

1. Build a resin wall.
2. Break it.

Seems to be a lighting bug from the regenerating resin walls PR according to this [person](https://github.com/tgstation/TerraGov-Marine-Corps/pull/3889#issuecomment-624967558). Not entirely sure if it is the cause but I can confirm the bug can be reproduced.

In [this video](https://youtu.be/fOAJqro7eZw) you can see near pitch black shadows of where the resin walls were. However there was one strange turf that seemed to only darken from the sides, shown close to the end of the video, where the cursor is next to the wooden barricade. Very likely something to do with lighting since it becomes more opaque the closer you move a light to it. It is not visible when I use the Hivelord at first since it has night vision. Setting the night vision to max removes the shadows from your view.

**Comment by psykzz** [comment]
This only happens when a xeno removes the wall, not when a marine does.

Calling `lighting_build_overlay` on the turf is a temp fix.

**Comment by GravitronBlue** [comment]
> This only happens when a xeno removes the wall, not when a marine does.

Far as I noticed during local testing the turf lighting is bugged the moment the thick resin wall is created, regardless I did some local debugging and possibly found the issue and maybe have a fix for it.

https://github.com/tgstation/TerraGov-Marine-Corps/pull/3889/files#diff-f073b96331988c10a03f54420a6e509cR229

```
	if(X.selected_resin == /turf/closed/wall/resin/regenerating)
		T.ChangeTurf(X.selected_resin)
		new_resin = T
```

Seems to only accepts a typepath for
`/turf/closed/wall/resin/regenerating`
and not the new
`/turf/closed/wall/resin/regenerating/thick`
used by hivelords.

I'm honestly unsure but I'm going to presume skipping over `T.ChangeTurf(X.selected_resin)` is what results with broken lighting.

Anyway local fix  I made was just to create another if statement following the previous one:
```
	if(X.selected_resin == /turf/closed/wall/resin/regenerating/thick)
		T.ChangeTurf(X.selected_resin)
		new_resin = T
```

Anyway if a coder could vertify what I've said that'd be great.
